### ISSUE TITLE
VSC not stopping on breakpoints in Azure Functions the last few days or so

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/ptvsd#1409]
@dxkaufman commented on [Fri May 03 2019](https://github.com/microsoft/vscode-python/issues/5547)

I apologize if I'm posting this in the wrong place; I'm quite new to python and vsc and git... 

## Environment data

- VS Code version: 1.33.1
- Extension version (available under the Extensions sidebar): 2019.4.12954
- OS and version: Mac OS X 10.13.6
- Python version (& distribution if applicable, e.g. Anaconda): 3.6.5
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): virtualenv (I think)
- Relevant/affected Python packages and their versions: N/A

## Expected behaviour

While attempting to test modifications to an Azure function (either timer or http trigger), if I set a break point on a line (not based on conditions, just break at line X), it it should break when it gets to that line.  Several days ago, this was working just fine.  As far as I'm aware, I haven't changed anything about the VSC or Python installation since then.  I'm afraid I didn't make a note of exactly when it stopped working; I think it was this week (on or after Mon 4/29, but it could have been sooner).

## Actual behaviour

It doesn't stop.

## Steps to reproduce:
1. open a working Azure time trigger function
2. in function.json, change the cron to something that will cause it to run soon (I usually check the time and set it to something about 30 seconds out)
3. set a break point in the code where you know the code will hit (for this test, just put it on the first code line in the main function)
4. run the code
5. if you get the same behavior I am, it'll just run right through the break point... 

Please note that I created a test.py file with nothing but a print statement (imports, no nothing), and when I put a breakpoint on that line and ran the program, it stopped on the breakpoint.

## Logs
Output for `Python` in the `Output` panel (`View`→`Output`, change the drop-down the upper-right of the `Output` panel to `Python`)

```
None relevant
```   

Output from `Console` under the `Developer Tools` panel (toggle Developer Tools on under `Help`; turn on source maps to make any tracebacks be useful by running `Enable source map support for extension debugging`)

```
None relevant
```

**Comment by int19h** [comment]
@dxkaufman Does it break for you if you manually call `breakpoint()` from that line? If it stops, what does the stack trace window show, particularly the paths?

**Comment by dxkaufman** [comment]
I can't seem to use it.  I'm still on 3.6.5, it looks as if I have to be on 3.7 for that to work?

**Comment by karthiknadig** [comment]
you can do this, which is equivalent:
```py
import ptvsd
ptvsd.break_into_debugger()
```

**Comment by dxkaufman** [comment]
Thanks, @karthiknadig.  @int19h, the answer is yes - if I call ptvsd.break_into_debugger() in my code, VSC goes into debug mode at that point.

**Comment by karthiknadig** [comment]
@dxkaufman Can you share the stack as requested by @int19h?

**Comment by dxkaufman** [comment]
@karthiknadig is this what you're asking for?  

![image](https://user-images.githubusercontent.com/26446314/57652341-306b9300-7584-11e9-9302-ea4a0a8dff75.png)

**Comment by karthiknadig** [comment]
@dxkaufman we will need the path to the files, in the stack frame. You can get them from the logs. Better yet just share the logs. You can enabled VSC side of the logs by setting `"logToFile": true` in launch json.

**Comment by dxkaufman** [comment]
OK, I set it up like this:

{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Attach to Python Functions",
      "type": "python",
      "request": "attach",
      "port": 9091,
      "preLaunchTask": "func: host start",
      "logToFile": true
    }
  ]
}

I found the attached debug.log file via the Open Extensions Folder.  I did not find any errors in the developer tools console window...

[debug97976.log](https://github.com/microsoft/ptvsd/files/3175592/debug97976.log)

I pulled this immediately after starting an azure function called KlipfolioAccess with ptvsd.break_into_debugger() as the first line of main

**Comment by int19h** [comment]
The paths in the log are `/usr/local/...`, so I think this might be the same as #1325.

If so, `"justMyCode": false` in launch.json configuration should help.

**Comment by dxkaufman** [comment]
It seems to be working again, at least as of yesterday...

**Comment by karthiknadig** [comment]
@dxkaufman Can you check if you are pulling pre-release version of `ptvsd`?

**Comment by dxkaufman** [comment]
@karthiknadig, it looks as if I have ptvsd 4.2.8.  That said, I'm not using it - when I said "it seems to be working again", I was referring to the built in breakpoints in VSC.  I used ptvsd when I needed it for a while, but I had to comment it out when I deployed to Azure (which I now realize was probably because I forgot to put it in my requirements file).  No idea what might have changed to make VSC's debugger start working on its own again.
